core/watcher/eventqueue/uuid.go

```diff
@@ -0,0 +1,135 @@
+// Any time rows change in the watched table, the changed UUIDs are emitted.
+type UUIDWatcher struct {
+	*BaseWatcher
+
+	out       chan []string
+	tableName string
+	selectAll string
+}
+
+// NewUUIDWatcher returns a new watcher that receives changes from the
+// input base watcher's db/queue when rows in the input table change.
+func NewUUIDWatcher(base *BaseWatcher, tableName string) *UUIDWatcher {
+	w := &UUIDWatcher{
+		BaseWatcher: base,
+		out:         make(chan []string),
+		tableName:   tableName,
+		selectAll:   fmt.Sprintf("SELECT uuid FROM %s", tableName),
+	}
+
+	w.tomb.Go(w.loop)
+	return w
+}
+
+// Changes returns the channel on which the UUIDs for
+// changed rows are sent to downstream consumers.
+func (w *UUIDWatcher) Changes() <-chan []string {
+	return w.out
+}
+
+func (w *UUIDWatcher) loop() error {
+	subscription, err := w.eventQueue.Subscribe(changestream.Namespace(w.tableName, changestream.All))
+	if err != nil {
+		return errors.Annotatef(err, "subscribing to namespace %q", w.tableName)
+	}
+	defer subscription.Unsubscribe()
+
+	changes, err := w.getInitialState()
+	if err != nil {
+		return errors.Annotate(err, "retrieving initial watcher state")
+	}
+
+	// By reassigning the in and out channels, we effectively ticktock between
+	// read mode and dispatch mode. This ensures we always dispatch deltas that
+	// we received before reading more, and every channel read/write is guarded
+	// by checks of the tomb and subscription liveness.
+	// We begin in dispatch mode in order to send the initial state.
+	var in <-chan []changestream.ChangeEvent
+	out := w.out
+
+	for {
+		select {
+		case <-w.tomb.Dying():
+			return tomb.ErrDying
+		case <-subscription.Done():
+			return ErrSubscriptionClosed
+		case subChanges, ok := <-in:
+			if !ok {
+				w.logger.Debugf("change channel closed for %q; terminating watcher", w.tableName)
+				return nil
+			}
+
+			// We have changes. Tick over to dispatch mode.
+			changes = transform.Slice(subChanges, func(c changestream.ChangeEvent) string { return c.ChangedUUID() })
+			in = nil
+			out = w.out
+		case out <- changes:
+			// We have dispatched. Tick over to read mode.
+			in = subscription.Changes()
+			out = nil
+		}
+	}
+}
+
+// getInitialState retrieves the current state of the world from the database,
+// as it concerns this watcher. It must be called after we are subscribed.
+// Note that killing the worker via its tomb cancels the context used here.
+func (w *UUIDWatcher) getInitialState() ([]string, error) {
+	parentCtx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	var uuids []string
+	err := w.db.Txn(w.tomb.Context(parentCtx), func(ctx context.Context, tx *sql.Tx) error {
+		rows, err := tx.QueryContext(ctx, w.selectAll)
+		if err != nil {
+			if err == sql.ErrNoRows {
+				return nil
+			}
+			return errors.Trace(err)
+		}
+		defer func() { _ = rows.Close() }()
+
+		for i := 0; rows.Next(); i++ {
+			uuids = append(uuids, "")
```

Comment:
This seems like a very strange construction. Why not use:
```
nextUUID := ""
rows.Scan(&nextUUID)
...
append(uuids, nextUUID)
```
It seems strange that you would put the uninitialized uuid into the list first.

It also simplifies the loop a bit because you don't need to track `i`

Reply:
We have the same pattern for structs, which is why it's probably like this. Might be worth swapping over as you're not allocating an extra slot if you don't need to prevent a growth on a slice in the error case (slices grow by doubling in certain case: https://go.dev/play/p/ZRI8ForB9UM).

Reply:
I'll change this.